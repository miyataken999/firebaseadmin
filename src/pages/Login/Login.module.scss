.errorMessage {
  margin: 0.7rem 0 0 0;
}
.socialButtons {
  flex-direction: column;
}
.icon {
  margin-right: 5px;
}
