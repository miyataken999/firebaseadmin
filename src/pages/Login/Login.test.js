import React from 'react';
import * as reactRedux from 'react-redux';

import * as actions from 'state/actions/auth';
import paths from '../Router/paths';
import Login from '.';

jest.mock('react-firebaseui');

describe('<Login /> rendering', () => {
  const dispatchMock = jest.fn();

  beforeEach(() => {
    jest
      .spyOn(reactRedux, 'useDispatch')
      .mockImplementation(() => dispatchMock);
    jest.spyOn(actions, 'authCleanUp').mockImplementation(jest.fn);
  });

  it('should render without crashing', () => {
    const { component } = renderWithProviders(<Login />)({
      auth: {
        userData: {},
      },
    });

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should display an error message when there is an error', () => {
    const { component } = renderWithProviders(<Login />)({
      auth: {
        userData: {},
        error: 'sample error',
      },
    });

    expect(component.container.querySelector('p.has-text-danger')).toBeTruthy();
  });

  it('should redirect to /home when the user is authenticated', () => {
    renderWithProviders(<Login />)({
      auth: {
        userData: {
          id: 'some userId',
        },
      },
    });

    expect(window.location.pathname).toBe(paths.ROOT);
  });
});
