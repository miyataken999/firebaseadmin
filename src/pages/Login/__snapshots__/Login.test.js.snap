// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> rendering should render without crashing 1`] = `
<DocumentFragment>
  <section
    class="section hero is-fullheight is-error-section"
  >
    <div
      class="hero-body"
    >
      <div
        class="container"
      >
        <div
          class="columns is-centered"
        >
          <div
            class="column is-two-fifths"
          >
            <div
              class="card has-card-header-background"
            >
              <header
                class="card-header"
              >
                <p
                  class="card-header-title"
                >
                  <span
                    class="icon"
                  >
                    <i
                      class="mdi mdi-lock default"
                    />
                  </span>
                  <span>
                    Login
                  </span>
                </p>
              </header>
              <div
                class="card-content"
              >
                <form>
                  <div
                    class="field"
                  >
                    <p
                      class="label"
                    >
                      E-mail Address
                    </p>
                    <div
                      class="control is-clearfix"
                    >
                      <input
                        class="input"
                        name="email"
                      />
                    </div>
                  </div>
                  <div
                    class="field"
                  >
                    <p
                      class="label"
                    >
                      Password
                    </p>
                    <div
                      class="control is-clearfix"
                    >
                      <input
                        class="input"
                        name="password"
                        type="password"
                      />
                    </div>
                  </div>
                  <br />
                  <div
                    class="field is-grouped"
                  >
                    <div
                      class="control"
                    >
                      <button
                        class="button is-black"
                        type="submit"
                      >
                        Login
                      </button>
                    </div>
                    <div
                      class="control"
                    >
                      <a
                        class="button is-outlined"
                        href="/recover-password"
                      >
                        Forgot Password?
                      </a>
                    </div>
                  </div>
                </form>
                <hr />
                <div
                  class="field is-grouped socialButtons"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
